sickbeard/providers/btn.py

```diff
@@ -73,78 +73,89 @@ def search(self, search_strings, age=0, ep_obj=None):  # pylint:disable=too-many
-                    # Note that these are individual requests and might time out individually.
-                    # This would result in 'gaps' in the results. There is no way to fix this though.
-                    if 'torrents' in parsed_json:
-                        found_torrents.update(parsed_json['torrents'])
+                for _, torrent_info in iteritems(found_torrents):
+                    (title, download_url) = self._get_title_url(torrent_info)
 
-            for _, torrent_info in iteritems(found_torrents):
-                (title, url) = self._get_title_and_url(torrent_info)
+                    if not all([title, download_url]):
+                        continue
 
-                if title and url:
-                    logger.log('Found result: {0}'.format(title), logger.DEBUG)
-                    results.append(torrent_info)
+                    seeders = torrent_info.get('Seeders', 1)
+                    leechers = torrent_info.get('Leechers', 0)
+
+                    # Filter unseeded torrent
+                    if seeders < min(self.minseed, 1):
+                        logger.log("Discarding torrent because it doesn't meet the "
+                                   "minimum seeders: {0}. Seeders: {1}".format
+                                   (title, seeders), logger.DEBUG)
+                        continue
+
+                    size = torrent_info.get('Size') or -1
+
+                    item = {
+                        'title': title,
+                        'link': download_url,
+                        'size': size,
+                        'seeders': seeders,
+                        'leechers': leechers,
+                        'pubdate': None,
+                        'hash': None,
+                    }
+                    logger.log('Found result: {0} with {1} seeders and {2} leechers'.format
+                               (title, seeders, leechers), logger.DEBUG)
+
+                    items.append(item)
+
+            results += items
 
-        # FIXME SORT RESULTS
         return results
 
     def _check_auth(self):
+
         if not self.api_key:
-            logger.log('Invalid api key. Check your settings', logger.WARNING)
+            logger.log('Missing or invalid api key. Check your settings', logger.WARNING)
+            return False
 
         return True
 
     def _check_auth_from_data(self, parsed_json):
 
-        if parsed_json is None:
-            return self._check_auth()
-
         if 'api-error' in parsed_json:
             logger.log('Incorrect authentication credentials: {0}'.format(parsed_json['api-error']), logger.DEBUG)
-            raise AuthException('Your authentication credentials for {0} are missing,'
-                                ' check your config.'.format(self.name))
+            return False
 
         return True
 
-    def _get_title_and_url(self, parsed_json):
+    def _get_title_url(self, parsed_json):
```

Comment:
They also have a point here. _get_title_and_url is inherited (or in this case overwritten). But _get_title_url is not. So this could be a @classmethod or @staticmethod.